docs/apps/components/checkbox.md

```diff
@@ -0,0 +1,198 @@
+|`this.refreshFieldDataSource`|Refresh the datasource data bound to the whole page|
+
+!!! Info
+    1. The show() and hide() methods can also be used to control the visibility of the checkbox component in response to user input.
+    2. **Fields** refers to a **component ID** or a **list of component IDs**. You can fetch the ID from the **Component Attribute** panel in the Page Builder.
+    3. Before using **this.show(fields)**, make sure the component is **hidden**. This can be done using **this.hide(fields)** or by **enabling** the **Hidden checkbox** in the Component Attribute panel.
+    4. Before using **this.enable(fields)**, make sure the component is **disabled**. This can be done using **this.disable(fields)** or by **enabling** the **Disabled checkbox** in the Component Attribute panel.
+
+#### Steps to use the methods for the Page Builder components
+
+1. Go to Form Attribute :material-menu-right: Action Panel :material-menu-right: Setting :material-menu-right: (Mounted | refresh | click 'Add action').
+2. Write the method/code as shown in the **Example** below.
+3. Click on `Save`.
+4. On the main screen click on `Save` again.
+5. Click on `Preview` to see the code in action.
+<img src= "/apps/components/img/check1.png">
+
+!!! Example
+    1. `this.addClassName(fields, className)`
+          * Go to Form Attribute :material-menu-right: Style Sheets :material-menu-right: add the class
+            ```
+            .abc{ // abc is the class name
+            background-color: red;
+            }
+            ```
+          * Follow the steps mentioned above, under Steps to use the methods for the Page Builder components
+          * ```
+            this.addClassName('checkbox_g0x4pt54z', 'abc')
+            ```
+
+    2. `this.removeClassName(fields, className)`
+        ```js
+        this.removeClassName('checkbox_g0x4pt54z', 'abc')
+        ```
+    
+    3. `this.getValues()`
+        ```js
+        var data = this.getValues();
+        console.log(data);
+        ```
+    
+    4. `this.hide(fields)`
+        ```js
+        var fields= ['checkbox_g0x4pt54z']
+        this.hide(fields)
+        ```
+    
+    5. `this.show(fields)`
+        ```js
+        var fields= ['checkbox_g0x4pt54z']
+        this.show(fields)
+        ```
+    
+    6. `this.disable(['fields'])`
+        ```js
+        this.disable(['checkbox_g0x4pt54z'])
+        ```
+    
+    7. `this.enable(['fields'])`
+        ```js
+        this.enable(['checkbox_g0x4pt54z'])
+        ```
+    
+    8. `this.getValue('fieldName')`
+        ```js
+        var checkname = this.getValue('checkbox_g0x4pt54z');
+        console.log('getValue', checkname);
+        ```
+    
+    9. `this.getComponent('component_ID')`
+        ```js
+        var checkname = this.getComponent('checkbox_g0x4pt54z');
+        console.log('getComponent', checkname);
+        ```
+    
+    10. `this.setData(Value)`
+        ```js
+        this.setData(
```

Comment:
Please fix the markdown styling here, it looks like you don't have closing tags or something llike that for the section that starts with `this.setData`

Reply:
done